### ISSUE TITLE
Recommend authentication with gh CLI rather than keys

### ISSUE THREAD

**Issue by klieret** [hsf-training/hsf-training-cicd-github#55]

**Comment by enibigir** [comment]
@klieret For GitHub Actions, I would suggest to update https://hsf-training.github.io/hsf-training-cicd-github/setup.html with the 2 options and let everyone choose what's convenient for the rest of the lessons

**Comment by wdconinc** [comment]
I was looking into this issue too. Is this in scope for this lesson, though? Shouldn't we make just refer to https://swcarpentry.github.io/git-novice/07-github/index.html for this?

**Comment by enibigir** [comment]
Thanks for digging in this issue. The question was raised in this lesson.
I completely agree to just add that reference with specific instructions

**Comment by klieret** [comment]
We could also point to @mambelli 's fork that contains the gh authentication.

I agree that this should be added to the carpentries lesson at some point though (but that's @mambelli 's decision)

**Comment by wdconinc** [comment]
Yes, I looked yesterday, but I didn't find @mambelli's fork at either https://github.com/mambelli/hsf-training-docker/branches or https://github.com/mambelli/hsf-training-singularity-webpage/branches. Where is it?

**Comment by klieret** [comment]
ah, no, he forked the git training IIRC (though I also don't find it at the moment)

**Comment by wdconinc** [comment]
Ah, https://github.com/mambelli/git-novice.

**Comment by wdconinc** [comment]
https://github.com/swcarpentry/git-novice/compare/gh-pages...mambelli:git-novice:gh-pages

**Comment by wdconinc** [comment]
But without a PR upstream, what is the plan here? https://github.com/swcarpentry/git-novice/pulls

**Comment by klieret** [comment]
We could just add a link to Marco's fork in our setup, right? Though honestly, I'm not sure how much sense that makes if we already put git in the prerequisites so that people will probably run through the carpentry git anyway....

**Comment by mambelli** [comment]
https://github.com/mambelli/git-novice is a fork of the sw carpentry git-novice. It is available at https://mambelli.github.io/git-novice/
I covered topics like some basic CI-CD and collaborating using forking and pull requests. SW carpentry is not interested in merging it because they don't like to extend the length of the current git-novice training. Even optional extensions are not of interest. If you think could be valuable I could separate the training and make a version with only changes in the existing episodes to see if they will accept that.

**Comment by wdconinc** [comment]
I don't think it is a maintainable model to (essentially) fork git-novice under hsf. I don't think swc will accept a separate training either. I think we may just have to turn this into a separate training ("git in HEP" or something) which we are going to have to maintain.

**Comment by mambelli** [comment]
Also for the authentication they objected that the SSH key is more general and that having both would extend the training.
I'm flexible about moving the extra material to a separate new or existing training (advanced git w/ both collaboration models and CI-CD?), but I feel strongly about keeping the GH CLI. I saw how much easier it was for the students. 
Let's discuss our preference next Monday and we can go back to sw carpentry w/ a new request.
